### ISSUE TITLE
Add mutation checks to API tests

### ISSUE THREAD

**Issue by acelaya** [shlinkio/shlink#844]
Infection makes use of code coverage reports in order to apply mutations and run the tests that affect mutated code.

With API tests, that is tricky, because covered code is running on a different process (the API server), while the process running the tests is just invoking that code through HTTP requests. Therefore, regular mechanisms to collect code coverage, cannot be used here.

API tests didn't even have proper coverage until very recently (see the PR where code coverage was added to API tests https://github.com/shlinkio/shlink/pull/845)

The problem is that, the code coverage tool is not properly collecting every covered line, but everything as a whole, therefore, generated coverage report is currently useless and needs to be fixed first.

> Description added after first two comments were placed. It is based on second one.

**Comment by umutphp** [comment]
Hi @acelaya ,

Is this issue open to contribution? If yes, I can be the assignee :).

A possible candidate to implement mutation check is [infection/infection](https://github.com/infection/infection) . Do you have any other package on your mind?

**Comment by acelaya** [comment]
Hey @umutphp. It would be amazing if you could contribute this, but it might be a bit trickier than it looks (I should have added a detailed description).

The project has three different test suits.

* Unit tests.
* Database tests (functional/integration tests run against a real database).
* API tests (E2E tests run against the REST API as a black box).

For the first two, there are mutation checks already, using infection.

The problem is that infection makes use of code coverage reports in order to apply mutations and run the tests that affect mutated code.

With API tests, that is tricky, because covered code is running on a different process (the API server), while the process running the tests is just invoking that code through HTTP requests. Therefore, regular mechanisms to collect code coverage, cannot be used here.

Because of that, API tests didn't even have proper coverage until very recently (see the PR where code coverage was added to API tests https://github.com/shlinkio/shlink/pull/845)

The problem is that, once I did that, I realized the code coverage tool was not properly collecting every covered line, but everything as a whole, therefore, generated coverage report is actually useless.

So, long story short, that needs to be fixed first, and then, mutation checks can be enabled for API tests through infection.

If you can find an alternative way to collect code coverage for API tests, that's also ok.

So, with all that in mind, if you still feel like doing it, the contribution will be more than welcome 🙂
